service/controller/v12/resource/instance/create.go

```diff
@@ -411,59 +414,59 @@ func (r *Resource) deleteDrainerConfig(ctx context.Context, customObject provide
 //     loop 5: worker 2 drained
 //     loop 6: worker 2 reimage
 //
-func (r *Resource) nextInstance(ctx context.Context, customObject providerv1alpha1.AzureConfig, instances []compute.VirtualMachineScaleSetVM, drainerConfigs []corev1alpha1.DrainerConfig, instanceNameFunc func(customObject providerv1alpha1.AzureConfig, instanceID string) (string, error), versionValue map[string]string) (*compute.VirtualMachineScaleSetVM, *compute.VirtualMachineScaleSetVM, *compute.VirtualMachineScaleSetVM, error) {
+func (r *Resource) nextInstance(ctx context.Context, customObject providerv1alpha1.AzureConfig, instances []compute.VirtualMachineScaleSetVM, drainerConfigs []corev1alpha1.DrainerConfig, instanceNameFunc func(customObject providerv1alpha1.AzureConfig, instanceID string) (string, error), versionValue map[string]string) (*workingSet, error) {
 	var err error
 
-	var instanceToUpdate *compute.VirtualMachineScaleSetVM
-	var instanceToDrain *compute.VirtualMachineScaleSetVM
-	var instanceToReimage *compute.VirtualMachineScaleSetVM
+	var ws *workingSet
 	{
 		r.logger.LogCtx(ctx, "level", "debug", "message", "looking for the next instance to be updated, drained or reimaged")
 
-		instanceToUpdate, instanceToDrain, instanceToReimage, err = findActionableInstance(customObject, instances, drainerConfigs, instanceNameFunc, versionValue)
+		emptyWorkingSet := workingSet{}
+
+		ws, err = getWorkingSet(customObject, instances, drainerConfigs, instanceNameFunc, versionValue)
 		if IsVersionBlobEmpty(err) {
 			// When no version bundle version is found it means the cluster just got
 			// created and the version bundle versions are not yet tracked within the
 			// parameters of the tenant cluster's VMSS deployment. In this case we
 			// must not select an instance to be reimaged because we would roll a node
 			// that just got created and is already up to date.
 			r.logger.LogCtx(ctx, "level", "debug", "message", "no instance found to be updated, drained or reimaged")
-			return nil, nil, nil, nil
+			return &emptyWorkingSet, nil
```

Comment:
```suggestion
			return nil, nil
```
With the latest `workingSet` implementation, it's safe to call its methods on nil value. Also see similar patterns later.

Reply:
done